src/lib/ext-activitylog.js

```diff
@@ -160,15 +198,28 @@ class Controller {
   }
 
   onFilterChange(filterObject) {
-    const { filterDetail, isFilterRemoved } = filterObject;
+    const { filterDetail, isFilterAdded } = filterObject;
 
-    this.model[isFilterRemoved ? 'removeFilter' : 'addFilter'](filterDetail);
+    this.model[isFilterAdded ? 'addFilter' : 'removeFilter'](filterDetail);
     this.view.setLogFilter((log) => this.isFilterMatched(log));
   }
+
+  onKeywordChange(newKeyword) {
+    this.model.setFilterKeyword(newKeyword);
+    this.view.setLogFilter((log) => this.isFilterMatched(log));
+  }
+
+  isFilterMatched(log) {
+    return this.model.matchLogWithFilterObj(log);
+  }
+
+  onFilterUpdate(filterObject) {
+    this.model.addFilter(filterObject);
```

Comment:
In isolation, this method seems to be missing a notification to re-apply filtering.

When I look up the caller, it turns out that this filter is only added when a new log entry is added (visible by default).

To clarify that, it would be more useful to reflect that in the method name (e.g. `onNewFilterOption`) and add a comment to explain why it is not necessary to trigger the log updater.

Reply:
Should I keep this method ? I will have to keep the `filterupdate` event then. I can also do the same thing with `filterchange` event. What do you suggest ?